filter/src/main/java/no/sb1/troxy/filter/AutoRegexNewRecording.java

```diff
@@ -141,6 +173,57 @@ protected void filterNewRecording(Recording recording) {
         responseTemplate.setContent(getReplacedField(responseTemplate.getContent(), responseContentPatterns, false));
     }
 
+    private String getReplacedFieldLineByLine(String input, List<PatternWithReplacement> patterns) {
+        // do nothing if no pattern
+        if (patterns.isEmpty()) {
+            return input;
+        }
+
+        // unescape input and chop off leading "^" and trailing "$"
+        String text = unescape(input.substring(1, input.length() - 1));
+        if (text == null) {
+            log.warn("Unable to unescape input '{}', can't search & replace text with regular expressions", input);
+            return input;
+        }
+
+        Set<Integer> linesToNotEscape = new HashSet<>();
+        String[] lines = text.split("\n");
+        for (PatternWithReplacement pattern : patterns) {
+            for (int currentLine = 0; currentLine < lines.length; currentLine++) {
+                String line = lines[currentLine];
+                Matcher matcher = pattern.getPattern().matcher(line);
+                if (matcher.find()) {
+                    String replacement = pattern.getReplacement();
+                    if (replacement == null) {
+                        final String patternAsString = matcher.pattern().pattern();
+                        lines[currentLine] = line.replaceAll(patternAsString, RequestPattern.escape(patternAsString));
```

Comment:
Mention that we replace inline so the next filter will get the output value of the current filter? That should be conveyed somehow.